redfish/virtualmedia.go

```diff
@@ -300,12 +291,26 @@ type VirtualMediaConfig struct {
 }
 
 // InsertMediaConfig sends a request to insert virtual media using the VirtualMediaConfig struct
-func (virtualmedia *VirtualMedia) InsertMediaConfig(config VirtualMediaConfig) error { //nolint
+func (virtualmedia *VirtualMedia) InsertMediaConfig(config VirtualMediaConfig) (*Task, error) { //nolint
 	if !virtualmedia.SupportsMediaInsert {
-		return errors.New("redfish service does not support VirtualMedia.InsertMedia calls")
+		return nil, errors.New("redfish service does not support VirtualMedia.InsertMedia calls")
+	}
+
+	resp, err := virtualmedia.PostWithResponse(virtualmedia.insertMediaTarget, config)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusAccepted {
```

Comment:
I'm a little worried some vendor implementations might just return a 200 here, just based on some of the oddities I've seen reported with some implementations. But I'm not suggesting this be changed, just noting it so if we run in to a report of an issue there is a breadcrumb here. :D

Reply:
Would a TODO or similar comment make more sense than history-in-issues?

Reply:
Yeah, not a bad idea!